cartography/intel/azure/iam.py

```diff
@@ -336,8 +336,12 @@ def get_roles_list(client: AuthorizationManagementClient) -> List[Dict]:
             result = client.role_definitions.get_by_id(role["role_definition_id"], raw=True)
             result = result.response.json()
             role['roleName'] = result['properties']['roleName']
-            role['permissions'] = result['properties']['permissions']
-
+            role['permissions'] = []
+            for action in result['properties']['permissions'][0]['actions']:
```

Comment:
use the get based approach and also, check for length of the `permissions` object before accessing the 0th element.